src/components/Identity/customerIds/createCustomerIds.js

```diff
@@ -2,19 +2,27 @@ import { normalizeCustomerIds, validateCustomerIds } from "./util";
 import {
   convertBufferToHex,
   convertStringToSha256Buffer,
-  clone
+  clone,
+  isEmptyObject
 } from "../../../utils";
 
 export default eventManager => {
-  const hash = (originalIds, normalizedIds) => {
+  const hash = (originalIds, normalizedIds, logger) => {
     const idNames = Object.keys(normalizedIds);
     const idsToHash = idNames.filter(idName => originalIds[idName].hashEnabled);
     const idHashPromises = idsToHash.map(id =>
-      convertStringToSha256Buffer(normalizedIds[id].id)
+      convertStringToSha256Buffer(normalizedIds[id].id.trim().toLowerCase())
     );
     return Promise.all(idHashPromises).then(hashedIds => {
       return hashedIds.reduce((finalIds, hashedId, index) => {
-        finalIds[idsToHash[index]].id = convertBufferToHex(hashedId);
+        if (!hashedId) {
+          delete finalIds[idsToHash[index]];
+          logger.warn(
+            `This browser doesn't support hashing, alloy couldn't hash ${idsToHash[index]}`
```

Comment:
No need to call out "alloy" here (we already prefix all messages). You could simplify it to something like `Unable to hash identity ${idsToHash[index]} due to lack of browser support.`

Also, isn't `idsToHash[index]` an object here? If so, I think it might be emitting `This browser doesn't support hashing, alloy couldn't hash [object Object]`. I don't know that printing the whole object is necessary anyway. Can we just print the actual ID or whatever is most identifiable to the user?

Reply:
Tests too please!

Reply:
I agree with the message.
`idsToHash` is an array and `idsToHash[index]` gives the namespace, i have verified it in Sandbox and it is like 
`This browser doesn't support hashing, alloy couldn't hash email`

Reply:
In that case it might make more sense to say "Unable to hash email identity due to lack of browser support." Your call though.